scripts/tests/test_zkai_attention_derived_d128_mlp_fusion_attribution_gate.py

```
@@ -0,0 +1,94 @@
+            gate.validate_route_gate(route_gate)
+
+    def test_read_json_rejects_non_finite_constants(self) -> None:
+        with tempfile.TemporaryDirectory(dir=gate.EVIDENCE_DIR) as temp_dir:
+            path = gate.pathlib.Path(temp_dir) / "nan.json"
+            path.write_text('{"bad": NaN}', encoding="utf-8")
+            with self.assertRaises(gate.MlpFusionAttributionError):
+                gate.read_json(path, 1024, "nan JSON")
+
+    def test_read_json_open_error_is_structured(self) -> None:
+        with mock.patch.object(gate.os, "open", side_effect=OSError("boom")):
+            with self.assertRaises(gate.MlpFusionAttributionError):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)